java/arcs/core/data/AnnotationParam.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+/** A class containing an annotation param. */
+sealed class AnnotationParam(strValue: String?, numValue: Int? = null, boolValue: Boolean? = null) {
+    data class Str(val strValue: String) : AnnotationParam(strValue)
+    data class Num(val numValue: Int) : AnnotationParam(null, numValue)
+    data class Bool(val boolValue: Boolean) : AnnotationParam(null, null, boolValue)
+
+    fun strValue(): String {
```

Comment:
I'd suggest either going with `Str`, `Num`, `Bool` subclasses, *or* offering these `strValue`, `numValue`, ... helper methods, but not both.

Subclasses are probably better:

```
val param: AnnotationParam = getParamFromSomewhere()
when (param) {
  is AnnotationParam.Str -> {
    println(param.strValue)
    // Accessing some other field, e.g. param.boolValue, is a compile error.
  }
  ...
}
```

If you have an instance of `AnnotationParam.Str`, you don't want to be able to accidentally call `.boolValue()`, it will crash.